### PR TITLE
Add FORCE_BUILD option to cmake

### PR DESCRIPTION
**Pull request by xavisolesoft**
This PR fixes the next issue the next issue:
Fixes: eProsima/Fast-DDS#4528

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
-    ${CMAKE_COMMAND}
-    --find-package
-    -DNAME=foonathan_memory
-    -DCOMPILER_ID=$<IF:$<BOOL:${WIN32}>,MSVC,GNU>
-    -DLANGUAGE=CXX
-    -DMODE=EXIST
-    -DCMAKE_FIND_DEBUG_MODE=ON
-  ERROR_QUIET
-  RESULT_VARIABLE _EXIT_CODE)
+option(FORCE_BUILD "Force this package build even is already installed in the system" OFF)
```
Comment (JesusPoderoso) [inline CMakeLists.txt]:
```suggestion
option(FOONATHAN_MEMORY_FORCE_VENDORED_BUILD "Force foonathan memory vendor package build even it is already installed in the system" OFF)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -1,16 +1,20 @@
...
+
+if(FORCE_BUILD EQUAL OFF)
```
Comment (JesusPoderoso) [inline CMakeLists.txt]:
```suggestion
if(NOT FOONATHAN_MEMORY_FORCE_VENDORED_BUILD)
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
Hi @xavisolesoft, thanks for your contribution!
These are a couple of suggestions:

File: CMakeLists.txt (hunk 2 above)
Comment (xavisolesoft) [reply CMakeLists.txt]: I committed all suggested changes

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,16 +1,20 @@
...
-    ${CMAKE_COMMAND}
-    --find-package
-    -DNAME=foonathan_memory
-    -DCOMPILER_ID=$<IF:$<BOOL:${WIN32}>,MSVC,GNU>
-    -DLANGUAGE=CXX
-    -DMODE=EXIST
-    -DCMAKE_FIND_DEBUG_MODE=ON
-  ERROR_QUIET
-  RESULT_VARIABLE _EXIT_CODE)
+option(FOONATHAN_MEMORY_FORCE_VENDORED_BUILD "Force foonathan memory vendor package build even it is already installed in the system" OFF)
```
Comment (JesusPoderoso) [inline CMakeLists.txt]:
```suggestion
option(FOONATHAN_MEMORY_FORCE_VENDORED_BUILD "Force foonathan memory vendor package build even if it is already installed in the system" OFF)
```

**Review by JesusPoderoso — CHANGES_REQUESTED** [review]
Hi @xavisolesoft, I missed a small nitpick here.
This PR goes in once applied!

**Comment by xavisolesoft** [comment]
> Hi @xavisolesoft, I missed a small nitpick here. This PR goes in once applied!

Fixed, thank you!

**Review by JesusPoderoso — APPROVED** [review]
Thanks @xavisolesoft for your contribution!

LGTM

**Pull request merged**